### DISCUSSION TITLE
new component won't appear in the auto completion when I'm using helix

### DISCUSSION THREAD

**Discussion by Dangerise** [slint-ui/slint#5150] [Q&A]
In vsc, when I'm writing slint files, new components will appear in the auto completions in my rust code since I save the slint files, but helix didn't work like that

**Comment by ogoffart** [comment]
Could you paste a small sample file, with a comment on where you expect the completion to happen?

**Reply by Dangerise** [reply]
any code, I think the problem is that ra won't always expand the macro,  even the  file has changed.

**Reply by ogoffart** [reply]
Are you using a `slint!` macro or a .slint file from a build script?

**Reply by hunger** [reply]
I think the issue is that helix only runs one LSP per file. So if you use the rust macro, that will be rust-analyzer and there is no code completion inside the slint! macro in helix as the slint-lsp is not run at all.

**Reply by Dangerise** [reply]
@ogoffart I tried both of them,  there is no suggestion appear even the `slint!` macro or slint files has changed

**Reply by hunger** [reply]
You can configure the helix editor to run slint for rust in `languages.toml`:

```
[[language]]
name = "rust"
language-servers = [ "rust-analyzer", "slint-lsp" ]
```

It finally works for me :-)

**Reply by Dangerise** [reply]
@hunger thanks, and it makes slint-lsp work in slint macro, but there is still another problem
